apps/ledger-live-mobile/src/screens/PostOnboarding/PostOnboardingHub.tsx

```diff
@@ -0,0 +1,244 @@
+  navigation,
+}) => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { lastActionCompleted, actionsState } = usePostOnboardingHubState();
+  const [popupOpened, setPopupOpened] = useState(true);
+  const { actionCompletedHubTitle, actionCompletedPopupLabel } =
+    lastActionCompleted ?? {};
+
+  const clearLastActionCompleted = useCallback(() => {
+    dispatch(clearPostOnboardingLastActionCompleted());
+  }, [dispatch]);
+
+  useEffect(
+    /**
+     * The last action context (specific title & popup) should only be visible
+     * the 1st time the hub is navigated to after that action was completed.
+     * So here we clear the last action completed.
+     * */
+    () => clearLastActionCompleted,
+    [clearLastActionCompleted],
+  );
+
+  useEffect(() => {
+    /**
+     * Necessary because this component doesn't necessarily unmount when
+     * navigating to another screen, so we need to open the popup again in case
+     * the user closed it and then completed a new action.
+     */
+    setPopupOpened(true);
+  }, [actionCompletedPopupLabel]);
+
+  const allowClosingScreen = useRef<boolean>(true);
+
+  const navigateToMainScreen = useCallback(() => {
+    allowClosingScreen.current = true;
+    navigation.replace(NavigatorName.Base, {
+      screen: NavigatorName.Main,
+    });
+  }, [navigation]);
+
+  const handleClosePopup = useCallback(() => {
+    setPopupOpened(false);
+  }, [setPopupOpened]);
+
+  /**
+   * At 0: regular screen
+   * At 0.5: opaque black/white backdrop drawn on top
+   * At 1: "finished" text drawn on top of backdrop
+   */
+  const animDoneValue = useSharedValue(0);
+
+  const allDone = useAllPostOnboardingActionsCompleted();
+
+  const animationTimeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const clearAnimationTimeout = useCallback(() => {
+    animationTimeout.current && clearTimeout(animationTimeout.current);
+  }, [animationTimeout]);
+
+  const triggerEndAnimation = useCallback(() => {
+    let dead = false;
+    const onAnimEnd = () => {
+      clearAnimationTimeout();
+      animationTimeout.current = setTimeout(() => {
+        navigateToMainScreen();
+      }, 3000);
+    };
+    animDoneValue.value = withDelay(
+      2000,
+      withTiming(1, { duration: 1500 }, finished => {
+        if (finished && !dead) {
+          runOnJS(onAnimEnd)();
+        }
+      }),
+    );
+
+    /**
+     * Preventing screen closing:
+     * - removing header right (close button)
+     * - disabling gestures (necessary for iOS)
+     * - adding listener to "beforeRemove" (necessary for Android as the back button would still close otherwise)
+     */
+    navigation.setOptions({ gestureEnabled: false, headerRight: () => null });
+    navigation.getParent()?.setOptions({ gestureEnabled: false });
+    allowClosingScreen.current = false;
+    navigation.addListener("beforeRemove", e => {
+      if (dead) return;
```

Comment:
instead of checking "dead" here, can't we do navigation.removeEventListener on the clean up function of the useEffect? Otherwise we'll be putting lots of event listeners there that don't do anything actually and never removing them.

Reply:
good point thanks 👍